large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -301,12 +309,27 @@ private BlobStorageClient createAmazonS3Client() {
         this.getAmazonEndpointOverride().ifPresent(clientBuilder::endpointOverride);
         this.getAmazonRegion().ifPresent(clientBuilder::region);
         this.getAmazonCredentialsProvider().ifPresent(clientBuilder::credentialsProvider);
+        this.getAmazonSdkHttpClientBuilderInstance()
+                .ifPresent(clientBuilder::httpClientBuilder);
         if (this.enableAmazonS3PathStyleAccess()) {
             clientBuilder.forcePathStyle(true);
         }
         return new AmazonS3Client(clientBuilder.build());
     }
 
+    private Optional<SdkHttpClient.Builder> getAmazonSdkHttpClientBuilderInstance() {
+        final Class<?> c = getClass(AbstractLargeMessageConfig.S3_SDK_HTTP_CLIENT_BUILDER_CONFIG);
+        if (c == null) {
+            return Optional.empty();
+        }
+        final Object o = Utils.newInstance(c);
+        if (!(o instanceof SdkHttpClient.Builder)) {
+            throw new RuntimeException(c.getName() + " is not an instance of " + SdkHttpClient.Builder.class.getName());
+        }else {
```

Comment:
```suggestion
        } else {
```
Please double check formatting